### ISSUE TITLE
CA1801: False report for unimplemented method with expression body and return value

### ISSUE THREAD

**Issue by gabbert** [dotnet/roslyn-analyzers#2846]
#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

ReviewUnusedParametersAnalyzer

#### Diagnostic ID

[CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801-review-unused-parameters)

#### Repro steps

Define method with unused parameter that returns a value and throws NotImplementedException using expression bodied syntax.

#### Code sample
```
// Error CA1801  Parameter value of method Foo is never used.Remove the parameter or use it in the method body.
public int Foo(int value) => throw new NotImplementedException();

// ok
public void Bar(int value) => throw new NotImplementedException();

// ok
public int Baz(int value)
{
    throw new NotImplementedException();
}

// Error CA1801  Parameter value of method Foo is never used.Remove the parameter or use it in the method body.
public int Foo2(int value) => throw new NotSupportedException();

// ok
public void Bar2(int value) => throw new NotSupportedException();

// ok
public int Baz2(int value)
{
    throw new NotSupportedException();
}
```

**Comment by mavasani** [comment]
Which version of package are you using? This might be a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/2761, which was fixed in https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.5-beta1.final

**Comment by gabbert** [comment]
I was using 2.6.3, but I just reproduced it in 2.9.5-beta1.final
